tests/integration/components/polaris-text-field-test.js

```diff
@@ -0,0 +1,222 @@
+        onChange=(action handleChange)
+      }}
+    `);
+
+    assert.equal(findAll(spinnerSelector).length, 1, 'it renders a polaris-text-field/spinner component');
+
+    await triggerEvent(spinnerUpButtonSelector, 'click');
+    assert.equal(this.get('value'), 2, 'numeric value is incremented when the `up` icon is clicked');
+
+    this.set('value', 5);
+
+    await triggerEvent(spinnerDownButtonSelector, 'click');
+    assert.equal(this.get('value'), 4, 'numeric value is decremented when the `down` icon is clicked');
+  });
+
+  test('it renders as a textarea when `multiline` is true', async function(assert) {
+    await render(hbs`
+      {{polaris-text-field
+        multiline=true
+      }}
+    `);
+
+    assert.equal(findAll(textFieldMultilineSelector).length, 1, 'it renders a textarea');
+  });
+
+  test('it renders an actionable link when a `labelAction` hash is present', async function(assert) {
+    this.setProperties({
+      labelActionText: label,
+      labelActionCalled: false
+    });
+
+    this.set('labelAction', () => {
+      this.set('labelActionCalled', true);
+    });
+
+    await render(hbs`
+      {{polaris-text-field
+        labelAction=(hash
+          content=labelActionText
+          onAction=labelAction
+        )
+      }}
+    `);
+
+    const labelButton = find(labelActionButtonSelector);
+    assert.ok(labelButton, 'it renders a button in the label');
+    assert.equal(labelButton.textContent.trim(), label, 'the label button displays the passed-in content');
+
+    await triggerEvent(labelButton, 'click');
+
+    assert.ok(this.get('labelActionCalled'), 'the label action is called when the label button is clicked');
+  });
+
+  test('it renders help text when `helpText` is present', async function(assert) {
+    this.set('helpText', helpText);
+
+    await render(hbs`
+      {{polaris-text-field
+        helpText=helpText
+      }}
+    `);
+
+    const helpTextNode = find(helpTextSelector);
+    assert.ok(helpTextNode, 'it renders help text element');
+    assert.equal(helpTextNode.textContent.trim(), helpText, 'help text value is correct');
+  });
+
+  test('it renders a prefix and suffix when `prefix` or `suffix` is present', async function(assert) {
+    this.setProperties({
+      prefix,
+      suffix
+    });
+
+    await render(hbs`
+      {{polaris-text-field
+        prefix=prefix
+        suffix=suffix
+      }}
+    `);
+
+    let prefixNode = find(textFieldPrefixSelector);
+    assert.ok(prefixNode, 'it renders the prefix element');
+    assert.equal(prefixNode.textContent.trim(), prefix, 'prefix value is correct');
+
+    let suffixNode = find(textFieldSuffixSelector);
+    assert.ok(suffixNode, 'it renders the suffix element');
+    assert.equal(suffixNode.textContent.trim(), suffix, 'suffix value is correct');
+  });
+
+  test('it renders as a connected component when `connectedLeft` or `connectedRight` is present', async function(assert) {
+    this.setProperties({
+      left,
```

Comment:
These should be components :wink:

Reply:
Well _now_ they should be, after making your updates 😛

Reply:
Just leaving this comment as an update to the above comments, we decided to keep the left and right connected attributes to accept both String or Component types to match React.